src/executor-group.jl

```diff
@@ -22,9 +22,12 @@ type DataParallelExecutorGroup <: AbstractExecutorGroup
   context :: Vector{Context}
   execs :: Vector{Executor}
 
-  data_shapes :: Dict{Symbol, Tuple{Vararg{Int}}}
-  label_shapes :: Dict{Symbol, Tuple{Vararg{Int}}}
+  data_shapes :: Vector{Tuple{Vararg{Int}}}
+  label_shapes :: Vector{Tuple{Vararg{Int}}}
```

Comment:
Any particular reason to go back to vectors from dicts?
We should just speed up the dict pathway to remove the splatting penalty.